Sources/Basic/TemporaryFile.swift

```diff
@@ -84,11 +87,13 @@ public final class TemporaryFile {
     ///            set, dir will be set to `/tmp/`.
     ///     - prefix: The prefix to the temporary file name.
     ///     - suffix: The suffix to the temporary file name.
+    ///     - deleteOnClose: Whether the file should get deleted when the instance is deallocated.
     ///
     /// - Throws: TempFileError
-    public init(dir: AbsolutePath? = nil, prefix: String = "TemporaryFile", suffix: String = "") throws {
+    public init(dir: AbsolutePath? = nil, prefix: String = "TemporaryFile", suffix: String = "", deleteOnClose: Bool = true) throws {
         self.suffix = suffix
         self.prefix = prefix
+        self.deleteOnClose = deleteOnClose
         // Determine in which directory to create the temporary file.
         self.dir = determineTempDirectory(dir)
```

Comment:
Would you please measure the typical cost of this `determineTempDirectory` on a local filesystem, in microseconds? I have a bad suspicion.

Reply:
The first time it actually needs to get the environment variables, it takes about 1ms. From then on, each call takes around 12 microseconds.